src/server/utilities/liteATIClickTracking/index.ts

```diff
@@ -0,0 +1,92 @@
+export default () => {
+  window.addEventListener('load', () => {
+    document.addEventListener('click', (event: MouseEvent) => {
+      const targetElement = event.target as HTMLElement;
+      if (targetElement?.tagName === 'A') {
+        event.stopPropagation();
+        event.preventDefault();
+
+        const atiURL = targetElement.getAttribute('data-lite-ati-tracking');
+
+        if (atiURL == null) {
+          return;
+        }
+
+        const currentAnchorElement = event.currentTarget as HTMLAnchorElement;
+        const nextPageUrl = currentAnchorElement?.href;
+
+        const {
+          screen: { width, height, colorDepth, pixelDepth },
+          innerWidth,
+          innerHeight,
+        } = window;
+        const now = new Date();
+        const hours = now.getHours();
+        const mins = now.getMinutes();
+        const secs = now.getSeconds();
+
+        // COOKIE SETTINGS
+        const cookieName = 'atuserid';
+        const expires = 397; // expires in 13 months
+        const cookiesForPage = `; ${document.cookie}`;
+        const atUserIdCookie = cookiesForPage.split(`; ${cookieName}=`);
+        let atUserIdValue = null;
+
+        if (atUserIdCookie.length === 2) {
+          const cookieInfo = atUserIdCookie.pop()?.split(';').shift();
+
+          if (cookieInfo) {
+            const decodedCookie = decodeURI(cookieInfo);
+            const user = JSON.parse(decodedCookie);
+            atUserIdValue = user.val;
+          }
+        }
+
+        if (!atUserIdValue && crypto.randomUUID) {
+          atUserIdValue = crypto.randomUUID();
```

Comment:
Similarly here, `crypto` may not be natively supported in Opera Mini. We have a polyfill from IE here: https://github.com/bbc/simorgh/blob/2662a70f40f6c80aaa3e44b771ebb5ac2391a594/src/poly/legacy.js#L15-L21 which Opera could be using.

Reply:
I've been working on a PR which replaces uuid with crypto - wonder if we could borrow the generateUUID code from there? https://github.com/bbc/simorgh/blob/38995753d1d339c9f1f937d896610094e9f0ea48/src/app/lib/utilities/getUUID/index.ts

Also, `crypto.randomUUID` doesn't work on http, only https, meaning it wouldn't work properly on local... not a massive issue I suppose?